app/Metadata/Extractor.php

```diff
@@ -477,4 +486,22 @@ public function extract(string $fullPath, string $kind): array
 
 		return $metadata;
 	}
+
+	/**
+	 * https://github.com/LycheeOrg/Lychee/issues/1200.
+	 *
+	 * Ensures the provided DateTime value is functional
+	 *
+	 * @param bool|\DateTime $d
```

Comment:
```suggestion
	 * @param \DateTime $d
```

Reply:
I originally intended for this function call to be used for both `false` and value checking, but I ended up being pressed for time and didn't get around to it before I had to leave.  Unless there is a reason to explicitly check for `false` _and_ execute the call, I'd prefer to just execute the call for both scenarios.

Reply:
Please don't. Using union types is bad in most cases and in this specific case it is not necessary at all. If the caller has not a valid `\DateaTime` object, the caller should already know that is obviously not a valid datetime.

The only reason why you added `bool` here is by co-incidents, due to a bad design decision of the exif extractor to return `false` on error instead of throwing a proper exception. This may change in the future (when I will find the time).

Reply:
Personally I think it being `false or \DateTime` is fine, as that is exactly how the upstream PHP `\DateTime` creation mechanisms behave, so I don't think that is worthy of a change.

Ultimately, though, either way works.